python/ray/ml/train/examples/lightgbm_example.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+from functools import partial
+from typing import Tuple
+
+import ray
+from ray.ml.predictors.integrations.lightgbm import LightGBMPredictor
+from ray.ml.train.integrations.lightgbm import LightGBMTrainer
+from ray.data.dataset import Dataset
+from ray.ml.result import Result
+from ray.ml.constants import TRAIN_DATASET_KEY
+from ray.ml.preprocessors import StandardScaler
+from sklearn.datasets import load_breast_cancer
+from sklearn.model_selection import train_test_split
+
+
+def prepare_data() -> Tuple[Dataset, Dataset, Dataset]:
+    data_raw = load_breast_cancer(as_frame=True)
+    dataset_df = data_raw["data"]
+    dataset_df["target"] = data_raw["target"]
+    train_df, test_df = train_test_split(dataset_df, test_size=0.3)
+    train_dataset = ray.data.from_pandas(train_df)
+    valid_dataset = ray.data.from_pandas(test_df)
+    test_dataset = ray.data.from_pandas(test_df.drop("target", axis=1))
+    return train_dataset, valid_dataset, test_dataset
+
+
+def train_xgboost(num_workers: int, use_gpu: bool = False) -> Result:
+    train_dataset, valid_dataset, _ = prepare_data()
+
+    # Scale some random columns
+    columns_to_scale = ["mean radius", "mean texture"]
+    preprocessor = StandardScaler(columns=columns_to_scale)
+
+    # LightGBM specific params
+    params = {
+        "objective": "binary",
+        "metric": ["binary_logloss", "binary_error"],
+    }
+
+    trainer = LightGBMTrainer(
+        scaling_config={
+            "num_workers": num_workers,
+            "use_gpu": use_gpu,
+        },
+        label_column="target",
+        params=params,
+        datasets={TRAIN_DATASET_KEY: train_dataset, "valid": valid_dataset},
+        preprocessor=preprocessor,
+        num_boost_round=100,
+    )
+    result = trainer.fit()
+    print(result.metrics)
+
+    return result
+
+
+def predict_xgboost(result: Result):
+    _, _, test_dataset = prepare_data()
+    this_checkpoint = result.checkpoint
+    predictor = LightGBMPredictor.from_checkpoint(this_checkpoint)
+
+    predicted_labels = (
+        test_dataset.map_batches(predictor.predict, batch_format="pandas")
```

Comment:
This doesn't generally work since predictor isn't necessarily serializable right? Also, this loads the predictor locally.